### DISCUSSION TITLE
show grid results after click top button

### DISCUSSION THREAD

**Discussion by teafmartins** [serenity-is/Serenity#6955] [Help] [answered]
Hello,
In a grid, I have several heavy queries and the filters need to be added first. I want to run the query and show results in the grid after the user clicks on a certain button at the top. How can I do this?

![image](https://github.com/serenity-is/Serenity/assets/17210232/b4a7c318-cd55-4b8c-83c9-d85c670d574c)

Thank you

**Comment by vooba** [comment]
Override the method getGridCanLoad() to return true after the button has been clicked.

**Comment by teafmartins** [comment, marked as answer]
Got it! thx


        protected getGridCanLoad() {

            if (this.runOnDemand == 0)
                return false;
            return true;
        }

      getButtons() {
                  var buttons = super.getButtons();
      
     
                  buttons.push({
                      title: 'Show Results',
                      cssClass: 'ShowResults-button',
                      icon: 'fa-chevron-circle-right',
                      separator: true,
                      onClick: () => {
                          this.runOnDemand = 1;
                          this.refresh();
                          this.runOnDemand = 0;
                      }
                  });
      
                  return buttons;
              }

**Comment by teafmartins** [comment]
But with this solution EXPORT button doesn't work. ..

**Reply by vooba** [reply]
I think, getGridCanLoad have to be true to send the export request to the backend.

**Comment by teafmartins** [comment]
Yes, that's it! thx
